src/main/java/org/apache/maven/io/util/WriterUtils.java

```diff
@@ -162,14 +162,26 @@ public static void replaceXpp3DOM( final Element parent, final Xpp3Dom parentDom
                 if ( parts.length > 1 )
                 {
                     final String nsId = parts[0];
-                    final String nsUrl = dm.getAttribute( "xmlns:" + nsId );
+                    String nsUrl = dm.getAttribute( "xmlns:" + nsId );
                     final String name = parts[1];
-
+                    if ( nsUrl == null )
+                    {
+                        nsUrl = parentDom.getAttribute( "xmlns:" + nsId );
+                    }
                     elem = factory.element( name, Namespace.getNamespace( nsId, nsUrl ) );
                 }
                 else
                 {
-                    elem = factory.element( dm.getName(), parent.getNamespace() );
+                    Namespace root = parent.getNamespace();
+                    for ( Namespace n : parent.getNamespacesInherited())
```

Comment:
I actually want a namespace without a prefix. A complete fallback might be the parent namespace. Otherwise what was happening was that the echo got the parent xmlns:
```
<ac:if xmlns:ac="antlib:net.sf.antcontrib">
                                 <ac:available file="${project.build.directory}/classes" type="dir"/>
                                 <ac:then>
                              <echo>Copying test classes.</echo>